source/solution_deploy/source/wait_provider.py

```diff
@@ -0,0 +1,59 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Custom resource provider that waits for a specified time, then returns success"""
+
+from os import getenv
+import json
+from logging import basicConfig, getLevelName, getLogger
+from time import sleep
+import cfnresponse
+
+basicConfig(level=getLevelName(getenv("LOG_LEVEL", "WARNING")))
+logger = getLogger(__name__)
+
+
+class InvalidRequest(Exception):
+    """Invalid wait request"""
+
+
+def wait_seconds(wait: float):
+    """Wait for `wait` seconds"""
```

Comment:
On a second thought, the convention may have been reasonable at a time where Python didn't have type hints, when our only way to document the expected input and return value was a doc string. But in any case I disagree with the idea to duplicate function name as a doc string. (1) Because it asks developers to create noise and (2) it adds a burden to extracting multiple short functions out of a big one, hence setting a negative incentive to write fewer, longer functions.

Reply:
I agree completely. I was just following my linter here. I should just uninstall pylint.